src/main/scala/graphql/schema/CandidateModeType.scala

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package basic.graphql.schema
+
+import basic.ObservingMode
+import basic.search.Constraints
+import cats.effect._
+import sangria.schema._
+
+object CandidateModeType {
+
+  def apply[F[_]: Effect]: UnionType[Unit] =
+    UnionType(
+      name        = "CandidateModeType",
+      description = Some("Candidate observing mode."),
+      types       = List(spectroscopy[F])
```

Comment:
We can encode sum types by using a GraphQL `union`. The only constraint here is that the union members must be concrete types (not interface types or other union types). Right now there's just spectroscopy but we will add at least imaging.

Reply:
The GraphQL type system supports union types. This document is pretty good. Scroll up to see the table of contents on the right. https://graphql.org/learn/schema/#union-types